### ISSUE TITLE
Cleanup sidebar explorer to separate actual code from configuration files/aux files

### ISSUE THREAD

**Issue by vscodenpa** [vscodenpa/testissues#5944]
With how many configuration files/auxiliary files are now typical for a web project, it would be nice to have a way to physically separate these items from my actual code. For example, here is a recent project where no folders that actuall have any code in them are open:

![image](https://user-images.githubusercontent.com/6005836/184038595-c716d2d6-4c29-4a11-91c1-155abab70c8b.png)

That's a pretty long list and a lot of visual noise. What I've done to try and combat this is nest all my config files with the fileNesting feature under my package.json. This is okay, but feels like I'm using the feature well past its intent. I also don't have everything in there like things such as .devcontaienr, .vscode, _templates for hygen.

These are all things I don't touch on a daily, weekly or even monthly basis. For my other developers in the project, I don't expect they will ever change any of these.

It would be great to expand on the idea of fileNesting to allow splitting up of a projects file into different categories. Being able to have a fake folder that has all my actual code in it, then another folder for all my environment and editor setup would be phenomenal.

Basically, I want to get it so I can easily collapse out of view the things I rarely touch and get my projects actual code more visible. 

I guess another way this could be looked at is having a fake folder that shows anything that has been hidden. Commonly hidden files could show up there too such as node_modules or .git for those once in a blue moon times you need to access it.
